idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,361 @@
+            def solids_energy_accumulation_constraints(b, t):
+                return b.solids_energy_accumulation[t] == \
+                    - sum(b.reactions[t].reaction_rate[r] *
+                          b.volume_solid[t] *
+                          b.reactions[t].dh_rxn[r]
+                          for r in b.config.reaction_package.
+                          rate_reaction_idx)
+
+        if self.config.energy_balance_type == EnergyBalanceType.none:
+            # Fix solids temperature to initial value for isothermal conditions
+            @self.Constraint(
+                    self.flowsheet().config.time,
+                    doc="Isothermal solid phase constraint")
+            def isothermal_solid_phase(b, t):
+                if t == b.flowsheet().config.time.first():
+                    return Constraint.Skip
+                else:
+                    return (
+                            b.solids[t].temperature ==
+                            b.solids[0].temperature)
+
+    def initialize(blk, state_args=None, outlvl=idaeslog.NOTSET,
+                   solver=None, optarg=None):
+        """
+        This is a general purpose initialization routine for simple unit
+        models. This method assumes a single ControlVolume block called
+        controlVolume, and first initializes this and then attempts to solve
+        the entire unit.
+        More complex models should overload this method with their own
+        initialization routines.
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                           package(s) to provide an initial state for
+                           initialization (see documentation of the specific
+                           property package) (default = {}).
+            outlvl : sets output level of initialization routine
+                 * 0 = Use default idaes.init logger setting
+                 * 1 = Maximum output
+                 * 2 = Include solver output
+                 * 3 = Return solver state for each step in subroutines
+                 * 4 = Return solver state for each step in routine
+                 * 5 = Final initialization status and exceptions
+                 * 6 = No output
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating which solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        """
+
+        # Set solver options
+        init_log = idaeslog.getInitLogger(blk.name, outlvl)
+        opt = get_solver(solver, optarg)  # create solver
+        opt.options = optarg
+
+        # ---------------------------------------------------------------------
+        # Initialize control volume block
+        init_log.info('Initialize Thermophysical Properties')
+
+        # Initialize gas_phase block
+        gas_phase_flags = blk.gas.initialize(
```

Comment:
You do not have a way to pass initial guesses to the property packages during initialization.

Reply:
See my comments above.

Reply:
@bpaul4 take a look at how things are setup in the MB to make the updates. Let me know if you need any help.

Reply:
I updated the initialization routine, thank you @andrewlee94 and @chineduobiora for your suggestions.